common/otel_metrics.go

```diff
@@ -25,31 +26,90 @@ import (
 	"go.opentelemetry.io/otel/metric"
 )
 
-const (
-	// IOMethod annotates the event that opens or closes a connection or file.
-	IOMethod = "io_method"
+var (
+	fsOpsMeter     = otel.Meter("fs_op")
+	gcsMeter       = otel.Meter("gcs")
+	fileCacheMeter = otel.Meter("file_cache")
 
-	// GCSMethod annotates the method called in the GCS client library.
-	GCSMethod = "gcs_method"
+	// Attribute Keys
+	// ioMethodKey specifies the I/O method attribute (e.g., opened, closed).
+	ioMethodKey = attribute.Key("io_method")
+	// gcsMethodKey specifies the name of the GCS method
+	gcsMethodKey = attribute.Key("gcs_method")
+	// fsOpKey specifies the FS operation like LookupInode, ReadFile etc.
+	fsOpKey = attribute.Key("fs_op")
+	// fsErrCategoryKey specifies the error category. The intention is to reduce the cardinality of FSError by grouping errors together.
+	fsErrCategoryKey = attribute.Key("fs_error_category")
+	// readTypeKey specifies the read operation like whether it's Sequential or Random
+	readTypeKey = attribute.Key("read_type")
+	// cacheHitKey specifies whether the read operation from file cache resulted in a cache-hit or miss.
+	cacheHitKey = attribute.Key("cache_hit")
+
+	fsOpsOptionCache,
+	readTypeOptionCache,
+	ioMethodOptionCache,
+	gcsMethodOptionCache,
+	cacheHitOptionCache,
+	cacheHitReadTypeOptionCache,
+	fsOpsErrorCategoryOptionCache sync.Map
+)
 
-	// FSOp annotates the file system op processed.
-	FSOp = "fs_op"
+func loadOrStoreAttrOption[K comparable](mp *sync.Map, key K, attrSetGenFunc func() attribute.Set) metric.MeasurementOption {
```

Comment:
why is this method taking a function as argument ? why can't we pass attribute.NewSet as param?